pkg/helmreconciler/pruning.go

```diff
@@ -40,6 +40,17 @@ func (h *HelmReconciler) Prune(all bool) error {
 	if err != nil {
 		allErrors = append(allErrors, err)
 	}
+	//prune namespace created by istio operator at last
```

Comment:
This is quite a dangerous operation since it will delete anything that the user created in the namespace too. At the very least, we should create a flag to guard this - but IMO it may be better to leave this step up to the user since it's not very onerous. Thoughts @howardjohn @sdake ?

Reply:
I am not as familiar with common operator patterns, but it seems dangerous to me. I think we should error on the side of NOT deleting stuff - will a user be more upset if they have an empty namespace around, or if we delete their critical applications?

Reply:
Generally speaking, we shouldn't be creating or deleting namespaces.  If we are creating namespaces, access should be restricted as, by definition, the operator is managing the lifecycle of that namespace.

Typically, an operator will create resources in the same namespace as the CR defining the install/behavior or in a completely segregated namespace, where all interaction would be through the CR.

Reply:
i agree, at the minimum this should be guarded with a flag... like you see a confirmation - do you really want to delete this github repo?

Also, how do you know if the namespace isn't there prior to install of the operator?

Reply:
if the namespace is created by operator, we will mark it via adding related operator lables and annotations, eg: `install.operator.istio.io/owner-generation`.

If the namespace is created prior to install of the operator, this PR will make sure operator will not touch the namespace(not adding related labels and annotations)

Reply:
> Generally speaking, we shouldn't be creating or deleting namespaces. If we are creating namespaces, access should be restricted as, by definition, the operator is managing the lifecycle of that namespace.
> 
> Typically, an operator will create resources in the same namespace as the CR defining the install/behavior or in a completely segregated namespace, where all interaction would be through the CR.

Assuming the Istio CR is creating deployments in the `istio-system` namespace Is this not handled by https://github.com/istio/operator/blob/master/deploy/operator.yaml#L33-L34?  If the answer is yes, perhaps there is some way to enforce the exact behavior we want as you have specified.